app/apphandlers/setup.go

```diff
@@ -242,5 +243,23 @@ func NewRsourcesService(deploymentType deployment.Type) (rsources.JobService, er
 		}
 	}
 
-	return rsources.NewJobService(rsourcesConfig)
+	service, err := rsources.NewJobService(rsourcesConfig)
+	if err != nil {
+		return nil, err
+	}
+	if rsourcesConfig.SharedConn != "" {
+		ctx := context.TODO()
+		lagStatus := service.MonitorLagLoop(ctx)
```

Comment:
can we do what we are doing now for the cleanupLoop? Don't start it by default, but let the caller decide when it will be started (so that we don't have 2 loops running, 1 in gateway and 1 in processor)